pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go

```diff
@@ -234,9 +234,31 @@ func (e *AzureLogAnalyticsDatasource) createRequest(ctx context.Context, dsInfo
 	return req, nil
 }
 
+// Error definition has been inferred from real data and other model definitions like
+// https://github.com/Azure/azure-sdk-for-go/blob/3640559afddbad452d265b54fb1c20b30be0b062/services/preview/virtualmachineimagebuilder/mgmt/2019-05-01-preview/virtualmachineimagebuilder/models.go
+type AzureLogAnalyticsAPIError struct {
+	Details *[]AzureLogAnalyticsAPIErrorBase `json:"details,omitempty"`
+	Code    *string                          `json:"code,omitempty"`
+	Message *string                          `json:"message,omitempty"`
+}
+
+type AzureLogAnalyticsAPIErrorBase struct {
+	Code       *string                      `json:"code,omitempty"`
+	Message    *string                      `json:"message,omitempty"`
+	Innererror *AzureLogAnalyticsInnerError `json:"innererror,omitempty"`
+}
+
+type AzureLogAnalyticsInnerError struct {
+	Code         *string `json:"code,omitempty"`
+	Message      *string `json:"message,omitempty"`
+	Severity     *int    `json:"severity,omitempty"`
+	SeverityName *string `json:"severityName,omitempty"`
+}
+
```

Comment:
> Looks like a number of these fields are not used in our code? e.g. Code, Severity.

Yes, we are not using them (at the moment?) but I wanted to reflect the real API response, that way people don't need to inspect the raw JSON in the future if they want use any of the other fields.

> Why do you need `omitempty` everywhere?

All of the parameters are in theory optional. I just used the Azure SDK as inspiration here (even though this specific API is not defined there): https://github.com/Azure/azure-sdk-for-go/blob/3640559afddbad452d265b54fb1c20b30be0b062/services/preview/virtualmachineimagebuilder/mgmt/2019-05-01-preview/virtualmachineimagebuilder/models.go

Reply:
Gotcha.

Removing those would simplify the code and make it more obvious.
Currently I see a risk it would:
* Misleadingly creates impression that use those fields, can be confusing to a future reader
* `omitempty` gives the impression that we are serializing these structs when it actually has no effect